File: app/src/main/java/be/vergauwen/simon/androidqualitycontrol/ui/ExampleComponent.java
package be.vergauwen.simon.androidqualitycontrol.ui;

import be.vergauwen.simon.androidqualitycontrol.core.di.ActivityScope;
import be.vergauwen.simon.androidqualitycontrol.core.di.ApplicationComponent;
import dagger.Component;

@ActivityScope
@Component(dependencies = ApplicationComponent.class)
public interface ExampleComponent  extends ExampleContract.Component<ExampleContract.View,ExamplePresenter>{}

File: app/src/main/kotlin/be/vergauwen/simon/androidqualitycontrol/ui/ExampleComponent.kt
package be.vergauwen.simon.androidqualitycontrol.ui

import be.vergauwen.simon.androidqualitycontrol.core.di.ActivityScope
import be.vergauwen.simon.androidqualitycontrol.core.di.ApplicationComponent
import dagger.Component

@ActivityScope
@Component(dependencies = arrayOf(ApplicationComponent::class))
interface ExampleComponent : ExampleContract.Component<ExampleContract.View,ExamplePresenter>